src/inference/dev_api/openvino/runtime/icore.hpp

```diff
@@ -170,6 +170,17 @@ class OPENVINO_RUNTIME_API ICore {
     virtual ov::SupportedOpsMap query_model(const std::shared_ptr<const ov::Model>& model,
                                             const std::string& device_name,
                                             const ov::AnyMap& config) const = 0;
+    /**
+     * @brief Get a transformed model of the input model with specified configuration
+     *
+     * @param model OpenVINO Model to be transformed
+     * @param device_name The target device name for which the transformation should be applied
+     * @param config Optional map of pairs: (config parameter name, config parameter value)
+     * @return A shared pointer to the transformed ov::Model object
+     */
+    virtual std::shared_ptr<ov::Model> get_transformed_model(const std::shared_ptr<const ov::Model>& model,
```

Comment:
It looks like GPU internal property. There is no need to add functions the such model can be get from device by internal property also.